common/lib/dependabot/pull_request_creator.rb

```diff
@@ -33,7 +33,8 @@ def initialize(source:, base_commit:, dependencies:, files:, credentials:,
                    branch_name_separator: "/", branch_name_prefix: "dependabot",
                    label_language: false, automerge_candidate: false,
                    github_redirection_service: "github-redirect.dependabot.com",
-                   custom_headers: nil, require_up_to_date_base: false)
+                   custom_headers: nil, require_up_to_date_base: false,
+                   work_item: nil)
```

Comment:
I wonder if it makes sense to start splitting up these PR creators based on the provider, there's more and more conditional logic / arguments, increasing the complexity and making it harder to add features and test things.

Not something that I'd expect in this PR, but more food for thought for ourselves cc @brrygrdn @feelepxyz

Reply:
Ah yeah would be ace to think about splitting these out. `work_item` here is slightly confusing as it's only for azure. Or could we add a generic `provider_metadata` hash that we pass to all pull request creators?

Reply:
@feelepxyz Totally agree with the idea. However, is this something you would like to see as part of this PR, or as a separate PR? It feels like a bit of a larger refactoring task in my opinion.

Reply:
@JamieMagee yeah lets leave a larger refactor out of this PR but thoughts on adding a new `provider_metadata` hash that gets passed in the Azure PR creator and then extract the `work_item` key?

Reply:
@feelepxyz I pushed an update that adds a `provider_metadata` hash currently just targeting the Azure specific properties, but this can easily be expanded to the other providers.

I think I can iterate on this a bit more. I think I'll pass `provider_metadata` directly through to the Azure client.

Reply:
Maybe I misunderstood, but what I was thinking, was that it would be better to to pass the Azure specific `work_item` to `Dependabot::PullRequestCreator`, but passing it to the Azure client explicitly is fine.

So my thinking was that we change `Dependabot::PullRequestCreator#initialize` to:

```ruby
    def initialize(source:, base_commit:, dependencies:, files:, credentials:,
                   branch_name_separator: "/", branch_name_prefix: "dependabot",
                   label_language: false, automerge_candidate: false,
                   github_redirection_service: "github-redirect.dependabot.com",
                   custom_headers: nil, require_up_to_date_base: false)
                   custom_headers: nil, require_up_to_date_base: false,
                   provider_metadata: {})
```

We'd then pass the `work_item` in that metadata:

```ruby
Dependabot::PullRequestCreator.new(
# ...
provider_metadata: { work_item: "my-work-item" }
)
```

And then when creating the Azure client we pull it out:

```ruby
    Azure.new(
        source: source,
        branch_name: branch_namer.new_branch_name,
        base_commit: base_commit,
        credentials: credentials,
        files: files,
        commit_message: message_builder.commit_message,
        pr_description: message_builder.pr_message,
        pr_name: message_builder.pr_name,
        author_details: author_details,
        labeler: labeler,
        work_item: provider_metadata[:work_item]
      )
```

That way we don't pollute the public interface of the generic PR creator with implementation specific arguments. @feelepxyz is that also what you were thinking?

Reply:
Ah, okay. I thought you want to just pass in an object and extract properties as needed 🤦. Just to clarify you want the public interface to have all common properties, and then a has for platform specific properties?